### PR TITLE
Added luacheck configuration to the project

### PR DESCRIPTION
**Pull request by RunsFor**
- Fix luacheck warnings
 - Eliminate use of global variables in tests
- Added basic .gitignore file for tarantool projects

### CODE REVIEW

File: graphql/parse.lua [hunk 1]
```diff
@@ -293,11 +317,11 @@ local graphQL = P {
...
 local function stripComments(str)
-  return (str .. '\n'):gsub('(.-\n)', function(line)
+  return (str .. '\n'):gsub('(.-\n)', function(_line)
     local index = 1
-    while line:find('#', index) do
-      local pos = line:find('#', index) - 1
-      local chunk = line:sub(1, pos)
+    while _line:find('#', index) do
+      local pos = _line:find('#', index) - 1
+      local chunk = _line:sub(1, pos)
```
Comment (Totktonada) [inline graphql/parse.lua:324]: Don't get this change.

File: graphql/rules.lua [hunk 2]
```diff
@@ -159,15 +158,17 @@ function rules.unambiguousSelections(node, context)
 
         validateField(key, fieldEntry)
       elseif selection.kind == 'inlineFragment' then
-        local parentType = selection.typeCondition and context.schema:getType(selection.typeCondition.name.value) or parentType
-        validateSelectionSet(selection.selectionSet, parentType)
+        local _parentType = selection.typeCondition
+          and context.schema:getType(selection.typeCondition.name.value)
+          or parentType
+        validateSelectionSet(selection.selectionSet, _parentType)
```
Comment (Totktonada) [inline graphql/rules.lua:164]: And here too.

File: graphql/types.lua [hunk 3]
```diff
@@ -201,7 +203,10 @@ end
 
 types.int = types.scalar({
   name = 'Int',
-  description = "The `Int` scalar type represents non-fractional signed whole numeric values. Int can represent values between -(2^31) and 2^31 - 1. ", 
+  description = [[
+    The `Int` scalar type represents non-fractional signed whole numeric values.
+    Int can represent values between -(2^31) and 2^31 - 1.
+  ]],
```
Comment (Totktonada) [inline graphql/types.lua]: This adds two newlines into the string. Is it by intention?

File: tests/parse.lua [hunk 4]
```diff
@@ -76,19 +79,19 @@ describe('parse', function()
     end)
 
     test('directives', function()
-      local operation = parse('query{}').definitions[1]
+      operation = parse('query{}').definitions[1]
```
Comment (Totktonada) [inline tests/parse.lua:82]: To be honest, I use redefinitions in my code and always disable this luacheck warning. It sometimes helps to keep code blocks being uniformly formatted.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+CMakeCache.txt
+cmake_install.cmake
+CMakeFiles
+CTestTestfile.cmake
+
+build
+build.luarocks
+
+# https://github.com/packpack/packpack
+packpack
```
Comment (Totktonada) [inline .gitignore:25]: Personally I prefer to add packpack just as a submodule.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# luarocks and tarantool rocks modules
```
Comment (Totktonada) [inline .gitignore:1]: I would move .gitignore to own commit: it is not related to luacheck.

Comment (Totktonada) [reply .gitignore:1]: And while we're here: I would add Vim swap files to .gitignore. See https://github.com/tarantool/graphql.0/blob/05f39946299cb2f35a97be326b992aace0205eaf/.gitignore

File: graphql/parse.lua (hunk 1 above)
Comment (RunsFor) [reply graphql/parse.lua:324]: There is a global [`line`](https://github.com/tarantool/graphql/blob/master/graphql/parse.lua#L5) variable, which was shadowed here.

File: graphql/rules.lua (hunk 2 above)
Comment (RunsFor) [reply graphql/rules.lua:164]: Same here. Would you suggest more appropriate change? Without refactoring. This project has not so much tests yet

File: graphql/parse.lua (hunk 1 above)
Comment (Totktonada) [reply graphql/parse.lua:324]: I would rename it to `lineno` or so and don't use one term 'line' for two different things within at least a file.

File: graphql/rules.lua (hunk 2 above)
Comment (Totktonada) [reply graphql/rules.lua:164]: I don't think that the warning about redefinition of a variable should be enabled by default: too much false positives (I mean, code where it is not a problem) and artificial changes like this one.

Comment (RunsFor) [reply graphql/rules.lua:164]: Tried to stick to the luacheck [tarantool standard](https://github.com/tarantool/luacheck/pull/8) here

File: graphql/types.lua (hunk 3 above)
Comment (RunsFor) [reply graphql/types.lua]: Thanks, made some "strips" and "gsubs" in the last commit

File: graphql/rules.lua (hunk 2 above)
Comment (Totktonada) [reply graphql/rules.lua:164]: Ugh. It seems that the decision was made, so redefinitions are banned. Okay.

Let's choose a meaningful name so. The value is a type from `on` clause if it exists. We can name it either `onConditionType` or `nextParentType`: one way is how it is obtained, another how it intended to be used. I think that the former way improves readability more then the latter.

**Comment by olegrok** [comment]
Fixed in scope of https://github.com/tarantool/graphql/pull/9

**Pull request closed without merging**